public/pages/Dashboard/Components/__tests__/AnomaliesLiveCharts.test.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { render, waitFor } from '@testing-library/react';
+import React from 'react';
+import {
+  AnomaliesLiveChart,
+} from '../AnomaliesLiveChart';
+import { selectedDetectors } from '../../../../pages/utils/__tests__/constants';
+import { Provider } from 'react-redux';
+import { coreServicesMock } from '../../../../../test/mocks';
+import { CoreServicesContext } from '../../../../components/CoreServices/CoreServices';
+import { mockedStore } from '../../../../redux/utils/testUtils';
+const anomalyResponse = [
+  {
+    ok: true,
+    response: {
+      anomaly_grade: 0.10949221682655441,
+      data_start_time: 1651817250642,
+      data_end_time: 1651817310642,
+      detector_id: 'gtU2l4ABuV34PY9ITTdm',
+      name: 'test2',
+    },
+  },
+];
+const anomalyResultQuery = [
+  {
+    anomaly_grade: 0.10949221682655441,
+    data_start_time: 1651817250642,
+    data_end_time: 1651817310642,
+    detector_id: 'gtU2l4ABuV34PY9ITTdm',
+  },
+];
+
+jest.mock('../../utils/utils', () => ({
+  getLatestAnomalyResultsForDetectorsByTimeRange: jest.fn(
+    () => anomalyResponse
+  ),
+  getFloorPlotTime: jest.fn(() => 1651817250642),
+  getLatestAnomalyResultsByTimeRange: jest.fn(() => anomalyResultQuery),
+  visualizeAnomalyResultForXYChart: jest.fn(),
+}));
+describe('<AnomaliesLiveChart /> spec', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
```

Comment:
If you're clearing all mocks before each unit test, can the mocks specified on line 37-44 be removed?

Reply:
good point, I'll test it out, should be able to be removed

Reply:
Actually on a second look, those lines have to be there in order to successfully mock this component. They are utility methods getting called in `AnomaliesLiveChart`. Those are what made the test finally work haha. `clearAllMocks` doesn't actually reset the implementation but just the usage data. I'll remove it for now as I only really added it as I saw it in other tests and not cause I understood why its needed. Will watch out for that more.